### PR TITLE
Fix and upgrade lapackpp

### PR DESCRIPTION
**Pull request by albestro**
I was not able to install `lapackpp` because it was failing at link-time due to misconfigured LAPACK libraries (in my particular instance I was using MKL).

Moreover, I took the chance to make available also the latest release branch.

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
@teonnik @Sely85 @G-Ragghianti @mgates3 can you review?

**Comment by mgates3** [comment]
Can you be more specific about the issue that you were encountering? The original code was successfully tested with MKL. CMake tends to prefer a list of absolute paths to libraries (e.g., `/dir1/liba.a;/dir2/libb.a;/dir1/liba.so`), which is what the original code was doing, vs. separating into ld_flags (e.g., `-L/dir1 -L/dir2 -la -lb`). If there is a problem to solve, a similar change may be needed in other packages (blaspp, slate).

**Comment by mgates3** [comment]
The new version 2020.10.02 was already merged into spack.

**Comment by albestro** [comment]
Sorry, I missed the open PR about the update, and I worked on it before the merge of #19587 into `develop`...so I just took the chance to update it. My fault.

About the error:

```shell
cmake ... '-DLAPACK_LIBRARIES=/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_intel_lp64.dylib;/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_sequential.dylib;/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_core.dylib;/usr/lib/libpthread.dylib;/usr/lib/libm.dylib;/usr/lib/libdl.dylib' ...
```

CMake was able to config LAPACK
```shell
...
Looking for LAPACK libraries and options
/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_intel_lp64.dylib\;/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_sequential.dylib\;/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_core.dylib\;/usr/lib/libpthread.dylib\;/usr/lib/libm.dylib\;/usr/lib/libdl.dylib  yes
Found LAPACK library: /opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_intel_lp64.dylib\;/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_sequential.dylib\;/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_core.dylib\;/usr/lib/libpthread.dylib\;/usr/lib/libm.dylib\;/usr/lib/libdl.dylib
...
```

but build failed during linking with this error
```
clang: error: no such file or directory: '/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_intel_lp64.dylib;/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_sequential.dylib;/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_core.dylib;/usr/lib/libpthread.dylib;/usr/lib/libm.dylib;/usr/lib/libdl.dylib'
```

The separation of the different elements of the array of libraries was problematic and it was escaped in the wrong way, IMHO.

Looking forward to hear from you if it is the right way to fix it in your opinion.

**Comment by albestro** [comment]
For completeness, this is what happens now. It seems to be still problematic.

```shell
cmake ... '-DLAPACK_LIBRARIES:STRING=-L/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib -L/usr/lib -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl' ...
```

```shell
...
-- Looking for LAPACK libraries and options
-L/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib\;-L/usr/lib\;-lmkl_intel_lp64\;-lmkl_sequential\;-lmkl_core\;-lpthread\;-lm\;-ldl  yes
Found LAPACK library: -L/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib\;-L/usr/lib\;-lmkl_intel_lp64\;-lmkl_sequential\;-lmkl_core\;-lpthread\;-lm\;-ldl
...
```

```
clang++ ... -L/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib;-L/usr/lib;-lmkl_intel_lp64;-lmkl_sequential;-lmkl_core;-lpthread;-lm;-ldl /Users/ialberto/spack/opt/spack/darwin-mojave-x86_64/apple-clang-10.0.1/blaspp-2020.10.01-nquozal76vgfs4mdf3euokt4i7e2qjsn/lib/libblaspp.dylib /usr/local/lib/libomp.dylib /opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_intel_lp64.dylib /opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_sequential.dylib /opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib/libmkl_core.dylib /usr/lib/libpthread.dylib /usr/lib/libm.dylib /usr/lib/libdl.dylib
ld: warning: directory not found for option '-L/opt/intel/compilers_and_libraries_2019.1.144/mac/mkl/lib;-L/usr/lib;-lmkl_intel_lp64;-lmkl_sequential;-lmkl_core;-lpthread;-lm;-ldl'
```

It seems it still needs some attention...

**Comment by G-Ragghianti** [comment]
Can we hold off on merging this until we are able to look into it in a bit more depth?  It will probably require some modification of the packages for blaspp and slate as well.

@albestro What is the full spec that you are building (including compiler)? Thanks.

**Comment by G-Ragghianti** [comment]
I think that the original problem is specific to the macos platform and is a problem within the SLATE build system.  As such, the "fix" should be put into the SLATE source rather than try to work around the OS-specific problems within the spack package.  SLATE is in intensive development right now, and MacOS has not been the primary targeted platform, but there are plans to address these problems.  If you still wish to implement a fix within the spack environment, it should be implemented as a patch to the slate source (and a similar patch submitted to SLATE at https://bitbucket.org/icl/slate/ .  Thanks.

**Pull request closed without merging**
